gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/RootProvider.java

```diff
@@ -522,6 +527,9 @@ protected void doStart() throws Exception {
         rootRouter.route(PATH_RESET_PASSWORD)
                 .failureHandler(resetPasswordFailureHandler);
 
+        rootRouter.route(HttpMethod.POST, "/passwordHistory")
+                .handler(new PasswordHistoryHandler(passwordHistoryService, userService, domain));
```

Comment:
IHMO, we should add the `clientRequestParseHandlerOptional` before calling the PasswordHistoryHandler in order to have access to the client during the password validation. That will allow us to use the right PasswordSettings.